src/main.cpp

```diff
@@ -13,55 +13,219 @@
+    fs::create_directories(models_output_dir / "models");
+    fs::create_directories(models_output_dir / "maps");
+
+    std::vector<std::string> extensions = { ".mdl", ".ani", ".phy", ".dx11.vtx", ".vvd", ".vvc" };
+    std::vector<std::string> ent_types = { "snd", "env", "spawn", "fx", "script" };
+
+    fs::path exe_path = get_executable_path();
+    fs::path rmdlconv_path = exe_path / "rmdlconv.exe";
+
+    // Function to copy a file from VPK to the output directory
+    auto copy_file_from_vpk = [&](const std::string& src_path, const fs::path& dst_path) {
+        FileHandle_t src_file = g_pFileSystem->Open(src_path.c_str(), "rb");
+        if (!src_file) {
+            fprintf(stderr, "Failed to open source file: %s\n", src_path.c_str());
+            return;
+        }
+
+        int file_size = g_pFileSystem->Size(src_file);
+        std::vector<char> buffer(file_size);
+        g_pFileSystem->Read(buffer.data(), file_size, src_file);
+        g_pFileSystem->Close(src_file);
+
+        std::ofstream dst_file(dst_path, std::ios::binary);
+        if (!dst_file) {
+            fprintf(stderr, "Failed to create destination file: %s\n", dst_path.string().c_str());
+            return;
+        }
+        dst_file.write(buffer.data(), file_size);
+        dst_file.close();
+
+        printf("Copied file: %s\n", src_path.c_str());
+        };
+
+    // Copy .ent files
+    for (const auto& ent_type : ent_types) {
+        std::string ent_file = "maps/" + map_name + "_" + ent_type + ".ent";
+        fs::path dst_path = models_output_dir / "maps" / (map_name + "_" + ent_type + ".ent");
+        copy_file_from_vpk(ent_file, dst_path);
+    }
+
+    // Process model files
+    for (const auto& model_path : model_paths) {
+        fs::path base_path = fs::path(model_path);
+
+        // Only process .mdl files
+        if (base_path.extension() != ".mdl") {
+            continue;
+        }
+
+        fs::path dst_base_path = models_output_dir / base_path;
+        fs::create_directories(dst_base_path.parent_path());
+
+        // Copy all related files first
+        for (const auto& ext : extensions) {
+            fs::path current_path = base_path;
+            current_path.replace_extension(ext);
+            std::string current_path_str = current_path.string();
+
+            fs::path dst_path = dst_base_path;
+            dst_path.replace_extension(ext);
+
+            copy_file_from_vpk(current_path_str, dst_path);
+        }
+
+        // Now process the .mdl file with rmdlconv
+        fs::path mdl_path = dst_base_path;
+        fs::path rmdlconv_out_dir = dst_base_path.parent_path() / "rmdlconv_out";
+
+        // Prepare command line
+        std::wstring cmdLine = L"\"" + rmdlconv_path.wstring() + L"\" -targetversion 53 -nopause -convertmodel \"" + mdl_path.wstring() + L"\"";
+
+        // Set up process information
+        STARTUPINFOW si = { sizeof(si) };
+        PROCESS_INFORMATION pi;
```

Comment:
these short variable names aren't very clear
confused `pi` with `PI` when I first read this bit
I'd also prefer if we could use an approach without windows types like `DWORD` etc.